src/nodes.coffee

```diff
@@ -5342,6 +5369,17 @@ makeDelimitedLiteral = (body, options = {}) ->
     when other     then (if options.double then "\\#{other}" else other)
   "#{options.delimiter}#{body}#{options.delimiter}"
 
+sniffDirectives = (expressions, {notFinalExpression} = {}) ->
+  index = 0
+  lastIndex = expressions.length - 1
+  while index <= lastIndex and expression = expressions[index]
```

Comment:
In general we try to avoid assignments in conditionals, as it looks like a typo. Can `expression = expressions[index]` be moved to the first line in the `while` block?

Reply:
Sure, moved into the `while` block